src/stores/search/MetaSearchStore.ts

```diff
@@ -0,0 +1,166 @@
+                if (tableDataEntryWithActiveClass) {
+                    tableDataEntryWithActiveClass.display = tableDataEntryWithActiveClass.display.replace(activeClass, '')
+                }
+
+                this.highlightedMetaData.forEach((entry) => {
+                    if (Array.isArray(entry.value)) {
+                        entry.value = entry.value.map((item) => (item as string).replace(activeClass, ''))
+                    } else {
+                        entry.value = String(entry.value).replace(activeClass, '')
+                    }
+                })
+
+                const tableDataEntryWithHighlightId = this.highlightedTableData.find((entry) => String(entry.display).includes(highlightId))
+                if (tableDataEntryWithHighlightId) {
+                    tableDataEntryWithHighlightId.display = tableDataEntryWithHighlightId.display.replace(
+                        highlightId,
+                        (match: string) => `${match} ${activeClass}`
+                    )
+                }
+
+                this.highlightedMetaData.forEach((entry) => {
+                    if (Array.isArray(entry.value)) {
+                        entry.value = entry.value.map((item) => (item as string).replace(highlightId, (match: string) => `${match} ${activeClass}`))
+                    } else {
+                        entry.value = String(entry.value).replace(highlightId, (match: string) => `${match} ${activeClass}`)
+                    }
+                })
+                
+                this.scrollToHighlight()
+            }
+        )
+    }
+
+    scrollToHighlight = () => {
+        const container = document.querySelector('.activeTab')
+        if (!container) return
+
+        const activeSearchResult = container.querySelector(`#highlight-${this.currentHighlightIndex}`)
+
+        if (activeSearchResult) {
+            const parentRect = container.getBoundingClientRect()
+            const resultRect = activeSearchResult.getBoundingClientRect()
+            const scrollOffset = resultRect.top - parentRect.top - parentRect.height / 2
+
+            container.scrollTo({
+                top: container.scrollTop + scrollOffset,
+                behavior: 'smooth',
+            })
+        }
+    }
+
+    getSearchResultsCount = () => {
+        return this.searchResults
+    }
+
+    getCurrentHighlightIndex = () => {
+        return this.currentHighlightIndex
+    }
+
+    nextSearchResult = () => {
+        this.currentHighlightIndex = (this.currentHighlightIndex + 1) % this.searchResults
+    }
+
+    previousSearchResult = () => {
+        this.currentHighlightIndex = (this.currentHighlightIndex - 1 + this.searchResults) % this.searchResults
+    }
+
+    search = async (query: string): Promise<void> => {
+        this.searchResults = 0
+        this.loading = true
+
+        try {
+            await this.searchSemaphore.acquire()
```

Comment:
I removed it after refactoring the unnecessary Promise/async from textSearchStore. The semaphore is no longer needed. 

I wanted to control the number ongoing promises allowed. But since that is no longer the case, it can be removed. 

I will probably look for another solution with webworkers to create async searching for the other 2, Meta and SearchStore